diff --git a/src/Form/ServerForm.php b/src/Form/ServerForm.php
index 56689c7..5bd713c 100644
--- a/src/Form/ServerForm.php
+++ b/src/Form/ServerForm.php
@@ -89,7 +89,13 @@ class ServerForm extends EntityForm {
     $schemas = array_map(function ($definition) {
       return $definition['name'] ?? $definition['id'];
     }, $this->schemaManager->getDefinitions());
-    $schema = ($formState->getUserInput()['schema'] ?? $server->get('schema')) ?: reset(array_keys($schemas));
+    if (array_key_exists('schema', $formState->getUserInput())) {
+      $form_schema = $formState->getUserInput()['schema'];
+    }
+    else {
+      $form_schema = FALSE;
+    }
+    $schema = ($form_schema ?? $server->get('schema')) ?: reset(array_keys($schemas));
